### PR TITLE
Linux .net 6 support added

### PR DESCRIPTION
**Pull request by stesee**

### CODE REVIEW

File: ClosedXML/ClosedXML.csproj [hunk 1]
```diff
@@ -34,23 +33,9 @@
 		<ContinuousIntegrationBuild Condition="'$(GITHUB_ACTIONS)' == 'true'">true</ContinuousIntegrationBuild>
 		<GeneratePackageOnBuild>True</GeneratePackageOnBuild>
 	</PropertyGroup>
-
 	<ItemGroup>
 		<None Include="docs\nugetReadme.md" Pack="true" PackagePath="\" />
 	</ItemGroup>
-
-	<PropertyGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' ">
-		<DefineConstants>TRACE;_NETSTANDARD_;_NETSTANDARD2_0_</DefineConstants>
```
Comment (rominator1983) [inline ClosedXML/ClosedXML.csproj:43]: i suppose you did check if those STANDARD-constants were in use?

File: ClosedXML/Excel/Drawings/IXLPicture.cs [hunk 2]
```diff
@@ -57,21 +58,21 @@ public interface IXLPicture : IDisposable
         /// <returns>A created copy of the picture.</returns>
         IXLPicture Duplicate();
 
-        Point GetOffset(XLMarkerPosition position);
+        SKPoint GetOffset(XLMarkerPosition position);
 
-        IXLPicture MoveTo(Int32 left, Int32 top);
+        IXLPicture MoveTo(float left, float top);
```
Comment (rominator1983) [inline ClosedXML/Excel/Drawings/IXLPicture.cs:63]: i am not entirely sure if that's a good idea. It seems to me that this is something that should be checked against the OpenXML standard. why did you change this to float?

File: ClosedXML/Excel/Drawings/XLPicture.cs [hunk 3]
```diff
@@ -14,39 +14,43 @@ namespace ClosedXML.Excel.Drawings
...
         {
-            var properties = typeof(ImageFormat).GetProperties(BindingFlags.Static | BindingFlags.Public);
-            FormatMap = Enum.GetValues(typeof(XLPictureFormat))
-                .Cast<XLPictureFormat>()
-                .Where(pf => properties.Any(pi => pi.Name.Equals(pf.ToString(), StringComparison.OrdinalIgnoreCase)))
-                .ToDictionary(
-                    pf => pf,
-                    pf => properties.Single(pi => pi.Name.Equals(pf.ToString(), StringComparison.OrdinalIgnoreCase)).GetValue(null, null) as ImageFormat
-                );
+            var properties = typeof(SKEncodedImageFormat).GetProperties(BindingFlags.Static | BindingFlags.Public);
```
Comment (rominator1983) [inline ClosedXML/Excel/Drawings/XLPicture.cs]: it seems like the properties-Variable is not used => remove

Comment (rominator1983) [reply ClosedXML/Excel/Drawings/XLPicture.cs]: It looks to me like this might have had something to do with flags enums (where there can be "Properties" with the same values) in System.Drawing. I honestly did not check. I suppose SKEncodedImageFormat is not a flags enum?

File: ClosedXML/Excel/Drawings/XLPicture.cs [hunk 4]
```diff
@@ -77,18 +82,22 @@ internal XLPicture(IXLWorksheet worksheet, Stream stream, XLPictureFormat format
...
-            if (bitmap == null) throw new ArgumentNullException(nameof(bitmap));
+            if (codec == null) throw new ArgumentNullException(nameof(codec));
             this.ImageStream = new MemoryStream();
-            bitmap.Save(ImageStream, bitmap.RawFormat);
-            ImageStream.Seek(0, SeekOrigin.Begin);
-            DeduceDimensionsFromBitmap(bitmap);
 
-            var formats = FormatMap.Where(f => f.Value.Guid.Equals(bitmap.RawFormat.Guid));
+            using (var bitmap = SKBitmap.Decode(codec))
+            using (var data = bitmap.Encode(codec.EncodedFormat, 70))
```
Comment (rominator1983) [inline ClosedXML/Excel/Drawings/XLPicture.cs]: magic value 70 has no explanation => const?

File: ClosedXML/Excel/Drawings/XLPicture.cs [hunk 5]
```diff
@@ -154,15 +163,15 @@ internal set
 
         public MemoryStream ImageStream { get; private set; }
 
-        public Int32 Left
+        public float Left
```
Comment (rominator1983) [inline ClosedXML/Excel/Drawings/XLPicture.cs:165]: same as before

File: ClosedXML/Excel/RichText/XLFormattedText.cs [hunk 6]
```diff
@@ -176,14 +225,17 @@ public bool Equals(IXLFormattedText<T> other)
...
         private IXLPhonetics _phonetics;
-        public IXLPhonetics Phonetics 
+
+        public IXLPhonetics Phonetics
         {
             get { return _phonetics ?? (_phonetics = new XLPhonetics(_defaultFont)); }
         }
 
-        public Boolean HasPhonetics { get { return _phonetics != null; } }
+        public Boolean HasPhonetics
```
Comment (rominator1983) [inline ClosedXML/Excel/RichText/XLFormattedText.cs:238]: could you as a next step make a PR for fixing all formatting issues  in one step?
this is quite hard to read doing all fixes where files are touched.

File: ClosedXML/Excel/XLWorkbook_Load.cs [hunk 7]
```diff
@@ -1136,17 +1135,17 @@ private void LoadDrawings(WorksheetPart wsPart, IXLWorksheet ws)
                         picture.Placement = XLPicturePlacement.FreeFloating;
 
                         if (spPr?.Transform2D?.Extents?.Cx.HasValue ?? false)
-                            picture.Width = ConvertFromEnglishMetricUnits(spPr.Transform2D.Extents.Cx, GraphicsUtils.Graphics.DpiX);
+                            picture.Width = ConvertFromEnglishMetricUnits(spPr.Transform2D.Extents.Cx, 72);
```
Comment (rominator1983) [inline ClosedXML/Excel/XLWorkbook_Load.cs:1138]: magic value 72 => constant?

File: ClosedXML/Excel/XLWorkbook_Load.cs [hunk 8]
```diff
@@ -1136,17 +1135,17 @@ private void LoadDrawings(WorksheetPart wsPart, IXLWorksheet ws)
...
 
                         if (spPr?.Transform2D?.Extents?.Cy.HasValue ?? false)
-                            picture.Height = ConvertFromEnglishMetricUnits(spPr.Transform2D.Extents.Cy, GraphicsUtils.Graphics.DpiY);
+                            picture.Height = ConvertFromEnglishMetricUnits(spPr.Transform2D.Extents.Cy, 72);
```
Comment (rominator1983) [inline ClosedXML/Excel/XLWorkbook_Load.cs:1141]: magic value 72 => constant? further occurrences down below

File: ClosedXML_Tests/Excel/Saving/SavingTests.cs [hunk 9]
```diff
@@ -250,7 +251,8 @@ public void CannotSaveAsOverwriteExistingReadOnlyFile()
                     };
 
                     // Assert
-                    Assert.Throws(typeof(UnauthorizedAccessException), saveAs);
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                        Assert.Throws(typeof(UnauthorizedAccessException), saveAs);
```
Comment (rominator1983) [inline ClosedXML_Tests/Excel/Saving/SavingTests.cs:255]: isn't there an exception on linux?

File: ClosedXML_Tests/Excel/Styles/ColorTests.cs [hunk 10]
```diff
@@ -28,7 +28,7 @@ public void ColorNotEqualOperatorInPlace()
         [Test]
         public void ColorNamedVsHTML()
         {
-            Assert.IsTrue(XLColor.Black == XLColor.FromHtml("#000000"));
+            Assert.That(XLColor.Black, Is.EqualTo(XLColor.FromHtml("#FF000000")));
```
Comment (rominator1983) [inline ClosedXML_Tests/Excel/Styles/ColorTests.cs:31]: is there an alpha channel?  why is there an alpha channel? what does FF alpha mean? transparent or non transparent?

File: ClosedXML_Tests/OleDb/OleDbTests.cs [hunk 11]
```diff
@@ -25,7 +26,19 @@ public void TestOleDbValues()
...
+                        connection.Open();
+                    }
+                    catch (System.InvalidOperationException excpetion)
+                    {
+                        if (excpetion.Message == "The 'Microsoft.ACE.OLEDB.12.0' provider is not registered on the local machine.")
+                        {
+                            Assert.Ignore("Install driver from https://www.microsoft.com/en-za/download/details.aspx?id=13255");
+                        }
+                        else
+                            throw;
```
Comment (rominator1983) [inline ClosedXML_Tests/OleDb/OleDbTests.cs:40]: no braces

File: ClosedXML_Tests/TestHelper.cs [hunk 12]
```diff
@@ -91,12 +96,19 @@ public static void RunTestExample<T>(string filePartName, bool evaluateFormulae
...
+                    {
+                        return;
+                    }
+
+                    if (string.IsNullOrEmpty(allowedDiff))
+                    {
+                        Assert.Fail(formattedMessage);
+                    }
+
+                    Assert.That(message, Is.EqualTo(allowedDiff));
```
Comment (rominator1983) [inline ClosedXML_Tests/TestHelper.cs]: there could be a better message stating that there is an allowedDiff using the form `Assert.That(message, Is.EqualTo(allowedDiff), "allowed diff bla bla");`

File: ClosedXML_Tests/Utils/ExcelDocsComparer.cs [hunk 13]
```diff
@@ -20,7 +20,11 @@ public static bool Compare(Stream left, Stream right, out string message)
             using (var leftPackage = Package.Open(left, FileMode.Open, FileAccess.Read))
             using (var rightPackage = Package.Open(right, FileMode.Open, FileAccess.Read))
             {
-                return PackageHelper.Compare(leftPackage, rightPackage, false, ExcludeMethod, out message);
+                bool v = PackageHelper.Compare(leftPackage, rightPackage, false, ExcludeMethod, out message);
+                if (v)
+
```
Comment (rominator1983) [inline ClosedXML_Tests/Utils/ExcelDocsComparer.cs:24]: line breaks

File: ClosedXML_Tests/Utils/StreamHelper.cs [hunk 14]
```diff
@@ -156,9 +157,41 @@ private static string RemoveIgnoredParts(this string s, Uri uri, Boolean ignoreC
             if (ignoreGuids)
                 s = RemoveGuids(s);
 
+            s = RemoveNonCodingXmlFormatDiff(s);
```
Comment (rominator1983) [inline ClosedXML_Tests/Utils/StreamHelper.cs]: return directly

File: ClosedXML_Tests/Utils/StreamHelper.cs [hunk 15]
```diff
@@ -156,9 +157,41 @@ private static string RemoveIgnoredParts(this string s, Uri uri, Boolean ignoreC
...
 
+        private static string RemoveNonCodingXmlFormatDiff(string s)
+        {
+            try
+            {
+                var original = XDocument.Parse(s);
+
+                var normalized = new XDocument(original);
+
+                string v = normalized.ToString();
```
Comment (rominator1983) [inline ClosedXML_Tests/Utils/StreamHelper.cs]: directly return

File: ClosedXML_Tests/Utils/StreamHelper.cs [hunk 16]
```diff
@@ -156,9 +157,41 @@ private static string RemoveIgnoredParts(this string s, Uri uri, Boolean ignoreC
...
+                return v;
+            }
+            catch (System.Xml.XmlException ex)
+            {
+                if (ex.Message.Contains("Data at the root level is invalid."))
```
Comment (rominator1983) [inline ClosedXML_Tests/Utils/StreamHelper.cs:176]: is this a good idea considering translations? I suppose System.Xml is translated.
The normal way to check for such problems is to iterate using an XmlReader

**Review by rominator1983 — COMMENTED** [review]
please have a look at my comments

File: ClosedXML/ClosedXML.csproj (hunk 1 above)
Comment (stesee) [reply ClosedXML/ClosedXML.csproj:43]: Yep. Are dead.

File: ClosedXML/Excel/Drawings/XLPicture.cs (hunk 3 above)
Comment (stesee) [reply ClosedXML/Excel/Drawings/XLPicture.cs]: Addressed: not used variable dropped

File: ClosedXML_Tests/TestHelper.cs (hunk 12 above)
Comment (stesee) [reply ClosedXML_Tests/TestHelper.cs]: $"Actual diff '{message}' differs to expected diff '{expectedDiff}', file '{resourcePath}'"

File: ClosedXML_Tests/Utils/StreamHelper.cs (hunk 16 above)
Comment (stesee) [reply ClosedXML_Tests/Utils/StreamHelper.cs:176]: well.... its test code. i wont put any effort into this.

File: ClosedXML/Excel/Drawings/IXLPicture.cs (hunk 2 above)
Comment (rominator1983) [reply ClosedXML/Excel/Drawings/IXLPicture.cs:63]: I just looked it up and found that coordinates are of the following type:
`  <xsd:simpleType name="ST_CoordinateUnqualified">
    <xsd:restriction base="xsd:long">
      <xsd:minInclusive value="-27273042329600"/>
      <xsd:maxInclusive value="27273042316900"/>
    </xsd:restriction>`

on the other hand when this is serialized everything walks through ConvertToEnglishMetricUnits to make EMU out of pixels. So the only "problem" is maybe some loss of precision when values of "number of pixels" (that is what left and top are basically) are very high. Which on the other hand is quite unrealistic and especially unproblematic.

So bottom line: this is not ideal but Ok for production use imho
